### PR TITLE
fix: take checkpoint_s3_uri and checkpoint_local_path in Framework class

### PR DESCRIPTION
**Pull request by ChuyangDeng**
Reference: 0301362690

*Issue #, if available:*
Framework estimators does not pass checkpoint_s3_uri or checkpoint_local_path to EstimatorBase class, thus not setting CheckpointConfig S3Uri and LocalPath in training job: https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/session.py#L388

*Description of changes:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have updated any necessary [documentation](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by ChoiByungWook** [comment]
Can you please provide a code snippet/example (preferably from the TensorFlow Estimator) of how the customer would use this introduced change?

I am confused about how the already existing "checkpoint_path" parameter in the TensorFlow estimator translates with "checkpoint_s3_uri".

Can you please prove that your change works? Either introduce an integ test or paste outputs for a custom test.

Thanks!

File: src/sagemaker/estimator.py [hunk 1]
```diff
@@ -1363,6 +1365,17 @@ def __init__(
                 authentication if they are provided; otherwise, python SDK will
                 try to use either CodeCommit credential helper or local
                 credential storage for authentication.
+            checkpoint_s3_uri (str): The S3 URI in which to persist checkpoints
```
Comment (ChoiByungWook) [inline src/sagemaker/estimator.py]: How would this work with https://github.com/aws/sagemaker-python-sdk/blob/152d177138f7dabe11c98938a66aeaab9acabfa7/src/sagemaker/tensorflow/estimator.py#L211

Comment (ChuyangDeng) [reply src/sagemaker/estimator.py]: Added an integ test to validate the training job API call has expected CheckpointConfig.

The checkpoint_path is deprecated in the tensorflow script mode:

`AttributeError: training_steps, evaluation_steps, requirements_file, checkpoint_path are deprecated in script mode. Please do not set checkpoint_path.` (https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/tensorflow/estimator.py#L338)

Comment (ChoiByungWook) [reply src/sagemaker/estimator.py]: How does this fare for users of the TensorFlow framework mode?

Comment (ChuyangDeng) [reply src/sagemaker/estimator.py]: For users of TensorFlow framework mode, passing `checkpoint_s3_uri` behaves the same as script mode. Its value get passed to EstimatorBase eventually and set the training args to CheckpointConfig and do multi-part uploads checkpointing.

**Pull request merged**